driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -220,8 +220,18 @@ public Statement saveQuery(T entity, Option... options) {
         }
     }
 
+    private static class AliasedMappedPropertyComparator implements Comparator<AliasedMappedProperty>
+    {
+        @Override
+        public int compare(AliasedMappedProperty o1, AliasedMappedProperty o2) {
+            String col1Name = o1.mappedProperty.getMappedName();
+            String col2Name = o2.mappedProperty.getMappedName();
+            return col1Name.compareTo(col2Name);
+        }
+    }
+
     private ListenableFuture<BoundStatement> saveQueryAsync(T entity, final EnumMap<Option.Type, Option> options) {
-        final Map<AliasedMappedProperty, Object> columnToValue = new HashMap<AliasedMappedProperty, Object>();
+        final Map<AliasedMappedProperty, Object> columnToValue = new TreeMap<AliasedMappedProperty, Object>(new AliasedMappedPropertyComparator());
```

Comment:
Small nit, instead of creating a new comparator each time, you could just have 1 static comparator, i.e.:

```java
private static final Comparator<AliasedMappedProperty> PROPERTY_COMPARATOR = new Comparator<AliasedMappedProperty>() {
    @Override
    public int compare(AliasedMappedProperty o1, AliasedMappedProperty o2) {
        String col1Name = o1.mappedProperty.getMappedName();
        String col2Name = o2.mappedProperty.getMappedName();
        return col1Name.compareTo(col2Name);
    }
}
```